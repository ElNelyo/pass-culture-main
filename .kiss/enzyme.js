// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
