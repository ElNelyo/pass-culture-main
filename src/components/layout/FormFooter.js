import React from 'react'
import PropTypes from 'prop-types'

const FormFooter = ({
  canCancel,
  canSubmit,
  cancelLabel,
  className,
  submitLabel,
  theme,
}) => (
  <footer
    className={`pc-theme-${theme} pc-footer pc-form-controls flex-end flex-columns flex-0 pl48 ${className}`}
  >
    <button
      type="reset"
      onClick={() => {}}
      disabled={!canCancel}
      className="no-border no-background no-outline no-select fs20 pr20 flex-1"
    >
      <span className="is-full-width is-block text-center">
        {cancelLabel}
      </span>
    </button>
    <button
      type="submit"
      disabled={!canSubmit}
      className="no-border no-background no-outline no-select fs20 pl20 mr12"
    >
      <span className="is-bold">
        {submitLabel}
      </span>
    </button>
  </footer>
)

FormFooter.defaultProps = {
  canCancel: true,
  canSubmit: false,
  cancelLabel: 'Annuler',
  className: '',
  submitLabel: 'Valider',
  theme: 'red',
}

FormFooter.propTypes = {
  canCancel: PropTypes.bool,
  canSubmit: PropTypes.bool,
  cancelLabel: PropTypes.string,
  className: PropTypes.string,
  submitLabel: PropTypes.string,
  theme: PropTypes.string,
}

export default FormFooter
