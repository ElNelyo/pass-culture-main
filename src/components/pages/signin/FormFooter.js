/* eslint
  react/jsx-one-expression-per-line: 0 */
import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'

import { FormFooterObject } from '../../../types'

const renderSubmitButton = cancel => (
  <button
    id="signin-submit-button"
    type="submit"
    disabled={cancel.disabled}
    className={`flex-1  ${cancel.className || ''}`}
  >
    <span className="is-block">{cancel.label}</span>
  </button>
)

const renderLinkButton = submit => (
  <Link
    id="signin-signup-button"
    to={submit.url}
    disabled={submit.disabled}
    className={`flex-1  ${submit.className || ''}`}
  >
    <span className="is-block">{submit.label}</span>
  </Link>
)

class FormFooter extends React.PureComponent {
  render() {
    const { cancel, className, submit } = this.props

    return (
      <footer
        className={`pc-final-form-footer dotted-top-2x-white py7 flex-0 flex-columns text-center items-center fs20 ${className}`}
      >
        {cancel && cancel.url && renderLinkButton(cancel)}
        <hr className="dotted-left-2x-white flex-0" />
        {submit && !submit.url && renderSubmitButton(submit)}
      </footer>
    )
  }
}

FormFooter.defaultProps = {
  cancel: null,
  className: '',
  submit: null,
}

FormFooter.propTypes = {
  cancel: FormFooterObject,
  className: PropTypes.string,
  submit: FormFooterObject,
}

export default FormFooter
