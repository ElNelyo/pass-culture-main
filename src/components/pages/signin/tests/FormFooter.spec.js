import React from 'react'
import { shallow } from 'enzyme'

import FormFooter from '../FormFooter'

describe('src | components | pages | signin | FormFooter', () => {
  describe.skip('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {
        cancel: null,
        className: 'fake className',
        submit: {
          url: 'fake url',
        },
      }

      // when
      const wrapper = shallow(<FormFooter {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('render', () => {
    describe('when cancel is defined and cancel url is present ', () => {
      it('should contain renderLinkButton', () => {
        // given
        const props = {
          cancel: {
            className: 'fake cancel className',
            disabled: false,
            label: 'fake cancel label',
            url: 'fake cancel url',
          },
          className: 'fake className',
        }

        // when
        const wrapper = shallow(<FormFooter {...props} />)
        const renderLinkButton = wrapper.find('Link').props()

        // then
        expect(renderLinkButton.id).toEqual('signin-signup-button')
      })
    })
    describe('when submit is defined and submit url is not present', () => {
      it('should contain renderSubmitButton', () => {
        // given
        const props = {
          className: 'fake className',
          submit: {
            className: 'fake submit className',
            disabled: false,
            label: 'fake submit label',
            url: '',
          },
        }

        // when
        const wrapper = shallow(<FormFooter {...props} />)
        const renderSubmitButton = wrapper.find('button').props()

        // then
        expect(renderSubmitButton.id).toEqual('signin-submit-button')
      })
    })
  })
})
