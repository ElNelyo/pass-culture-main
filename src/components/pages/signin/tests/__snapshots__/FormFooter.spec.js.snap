// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | signin | FormFooter snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormFooter
    cancel={null}
    className="fake className"
    submit={
      Object {
        "url": "fake url",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <hr
          className="dotted-left-2x-white flex-0"
        />,
        false,
      ],
      "className": "pc-final-form-footer dotted-top-2x-white py7 flex-0 flex-columns text-center items-center fs20 fake className",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "dotted-left-2x-white flex-0",
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      false,
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <hr
            className="dotted-left-2x-white flex-0"
          />,
          false,
        ],
        "className": "pc-final-form-footer dotted-top-2x-white py7 flex-0 flex-columns text-center items-center fs20 fake className",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dotted-left-2x-white flex-0",
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        false,
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
